soql-standalone-parser/src/main/scala/com/socrata/soql/parsing/AbstractParser.scala

```diff
@@ -9,7 +9,7 @@ import com.socrata.soql.ast._
 import com.socrata.soql.environment.{ColumnName, FunctionName, TableName, TypeName}
 
 object AbstractParser {
-  class Parameters(val allowJoins: Boolean = true)
+  class Parameters(val allowJoins: Boolean = true, val systemColumnAliasesAllowed: Set[String] = Set.empty)
```

Comment:
Can this be a `Set[ColumnName]` instead?  Only because user code will probably have them in `ColumnName` form already, and moving the `ColumnName(name)` up two lines down below in `namedSelection` is simpler than `systemColumns.map(_.underlying)`